File: 643-maximum-average-subarray-i/maximum-average-subarray-i.cpp
class Solution {
public:
    double findMaxAverage(vector<int>& nums, int k) {
    double avg=-INFINITY;
    int l,r,sum=0;
    for(l=r=0;r<nums.size();r++){
        sum+=nums[r];
        if(r-l+1==k){
            double  a=(double)sum/k;
            avg=max(avg,a);
            sum-=nums[l++];
        }
    }
    return avg;
    }
};

File: 643-maximum-average-subarray-i/maximum-average-subarray-i.java
class Solution {
    public double findMaxAverage(int[] nums, int k) {
        int l=0;
        double sum=0;
        double avg=0;
        double maxsum=Double.NEGATIVE_INFINITY;
        for(int r=0;r<nums.length;r++){
            sum+=nums[r];
            if(r-l+1==k){
                maxsum=Math.max(maxsum,sum);
                avg=maxsum/k;
            
            sum-=nums[l];
            l++;
            }
        }
        return avg;
    }
}